tools/statplot.py

```diff
@@ -1,229 +1,305 @@
+        self.connect('key-press-event', self.KeyPressed)
+        self.set_border_width(8)
+        self.set_default_size(1600, 900)
+        self.set_position(Gtk.WindowPosition.CENTER)
+        self.set_title(statfile[-1])
+        self.statfile = statfile
+        self.vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=8)
+        self.vbox.set_homogeneous(False)
+        self.add(self.vbox)
+        hbox = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=8)
+        hbox.set_homogeneous(True)
+        self.vbox.pack_end(hbox, False, False, 0)
+        self.entries = self.ReadData(statfile)
+        self.xCombo = Gtk.ComboBoxText.new_with_entry()
+        self.xCombo.set_wrap_width(3)
+        self.yCombo = Gtk.ComboBoxText.new_with_entry()
+        self.yCombo.set_wrap_width(3)
+        self.PopulateCombo('load')
+        self.IniCompl(self.xCombo)
+        self.IniCompl(self.yCombo)
+        self.xCombo.connect('changed', self.ComboChangedX)
+        self.xCombo.connect('key-release-event', self.ReleaseFocus)
+        self.yCombo.connect('changed', self.ComboChangedY)
+        self.yCombo.connect('key-release-event', self.ReleaseFocus)
+        hbox.pack_start(self.xCombo, True, True, 0)
+        hbox.pack_start(self.yCombo, True, True, 0)
+        self.fig, self.ax = plt.subplots(nrows=1, ncols=1, num=0,
+                                         figsize=(14, 7))
+        self.canvas = FigureCanvas(self.fig)
+        self.canvas.set_can_focus(True)
+        self.vbox.pack_start(self.canvas, True, True, 0)
+        self.PlotType = 'line'
+        self.PlotData('create', self.PlotType, 'linear', 'linear')
+        self.toolbar = NavigationToolbar(self.canvas, self)
+        self.vbox.pack_start(self.toolbar, False, False, 0)
+
+    def ComboChangedX(self, widget):
+        complOptions = self.PopulateCompletion(self.xCombo.get_active_text())
+        dynModel = Gtk.ListStore.new([str])
+        for option in complOptions:
+            dynModel.append([option])
+        self.xCombo.get_child().get_completion().set_model(dynModel)
+        if self.xCombo.get_active_text() in sorted(self.entries.Paths()):
+            self.PlotData('update', self.PlotType, 'linear',
+                          self.ax.get_yscale())
+            self.canvas.grab_focus()
+
+    def ComboChangedY(self, widget):
+        complOptions = self.PopulateCompletion(self.yCombo.get_active_text())
+        dynModel = Gtk.ListStore.new([str])
+        for option in complOptions:
+            dynModel.append([option])
+        self.yCombo.get_child().get_completion().set_model(dynModel)
+        if self.yCombo.get_active_text() in sorted(self.entries.Paths()):
+            self.PlotData('update', self.PlotType, self.ax.get_xscale(),
+                          'linear')
+            self.canvas.grab_focus()
+
+    def FormatAxis(self, ax2fmt, scale):
+        if ax2fmt == 'x':
+            self.ax.set_scale = self.ax.set_xscale
```

Comment:
Recommend making this local, not modifying the `ax` object.

Reply:
Sensible, thank you.